src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,37 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeption.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new ValidationException("User is null");
+        }
+        if (user.getLogin() == null) {
+            throw new ValidationException("Login can't be null");
+        }
+        if (user.getLogin().length() < MIN_LENGTH) {
+            throw new ValidationException("Login length is less than 6 characters");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationException("User with such login exists");
+        }
+        if (user.getAge() == null || user.getAge() < MIN_AGE) {
+            throw new ValidationException("Age of user is less than 18 years");
+        }
```

Comment:
In login and password case you separated check for null and for "less than", be consistent.